src/hal/user_comps/mb2hal/mb2hal_HOWTO.ini

```diff
@@ -133,6 +134,16 @@ PIN_NAMES=cycle_start,stop,feed_hold
 #     mb2hal.m.n.int     (input)
 #     NELEMENTS needs to be 1 or PIN_NAMES must contain just one name.
 #     Both pin values are added and limited to 65535 (UINT16_MAX). Normally use one and let the other open (read as 0).
+#     Usage tip:   If you need to change individual bits in a register, use the MUX4 function.
+#                  # setup Modbus code 
+#                  # for control spindle driver
+#                  setp mux4.0.in0 1    # bit1-0 (01B:Stop)
+#                  setp mux4.0.in1 18   # bit1-0 (10B:Run) & bit5-4 (01B:FWD)
+#                  setp mux4.0.in2 34   # bit1-0 (10B:Run) & bit5-4 (10B:REV)
+#                  # RUN FOWARD, RUN REVERSE, STOP
+#                  net spindle-fwd 	spindle.0.forward => mux4.0.sel0
+#                  net spindle-rev 	spindle.0.reverse => mux4.0.sel1
+#                  net spindle-command	mb2hal.m.n.float <=  mux4.0.out
```

Comment:
First, the description is not correct. If you want to change individual bits you can use e.g. [`bitwise`](https://linuxcnc.org/docs/2.9/html/man/man9/bitwise.9.html)
What you are doing is you assign some specific values to one output signal depending on some bits. And the MUX output type is float, so not the best type if you want to have a bit-correct output.
And I see no relation here to mb2hal. This is more general.

Reply:
1) If I want use mb2hal for control spindle driver. I need change only bits in one register.
![Delta_C2000-manual_modbus](https://user-images.githubusercontent.com/96618597/222684231-10b830ea-5139-4736-a428-9fdab71f153d.png)

In ideal world producers of drivers should use for command START, STOP, etc, "fnct_05_write_single_coil(05 = 0x05) " It should be fantastic. But we dont live in Ideal world, so I invent some solution.

2) I am not opposed to a better solution for controlling Modbus drivers using mb2hal. I don't know how the HAL module bitwise would help me. 
Currently, the HAL component classicladder or the HAL driver designed for a specific driver is used to control the Modbus driver. Using the classiclader component to control the spindle driver feels like using a machine gun on a sparrow. Writing a HAL component for every driver that exists is not the solution.

3) This combination of mb2hal and MUX already works for me in two cases.

4) I agree that FLOAT type MUX.OUT is not ideal.

Reply:
>     4. I agree that FLOAT type MUX.OUT is not ideal.

I think `mux_generic` is better here.

However, this fits better somewhere in the driver section or in HAL howto than here.

Reply:
I think Modbus is most commonly used on CNC machines to control spindle drivers. That's why I put it here. If mb2hal was a component in a 3D printer project, I wouldn't put it there. If you don't agree with the example, I would like to provide information here that it is possible to use the ModBus function6 to control individual bits in the register.

Or we can generalize the example more.

Something similar is listed by the manufacturer of Delta drivers for input control:
![Delta_C2000-control_inputs](https://user-images.githubusercontent.com/96618597/222740353-f6db6cad-3ef5-497e-8b66-1738d709a342.png)


When I tried to get the first ModBus spindle driver running. I tried changing individual bits in the registry with fnct_05_write_single_coil, which of course didn't work.

mux_generic is better.  In the example, it would be enough to rewrite mux4 as mux-gen

Reply:
I suggest to write a short section in the [mb2hal doc page](http://linuxcnc.org/docs/2.9/html/drivers/mb2hal.html) or in the Hardware Examples section and add a note/reference in the sample INI file. But I don't want to bloat that INI file with general information.
A general howto would be nice how to set up linuxcnc with mb2hal to control VFDs.

Reply:
So, this part we will not added here. What should I do now?

Reply:
This is just my opinion. But if nobody else jumps in here it will probably not be added.
You can write a short tutorial if you think this is helpful for other users.